docs/src/pages/docs/guides/04-recruiter.mdx

```diff
@@ -123,28 +122,47 @@ You now combine the steps to implement branching logic based on the candidate’
 
 ```typescript
 const candidateWorkflow = new Workflow({
-  name: 'candidate-workflow',
+  name: "candidate-workflow",
   triggerSchema: z.object({
     resumeText: z.string(),
   }),
 });
 
 candidateWorkflow
-  .step(gatherCandidateInfo)
+  .step(gatherCandidateInfo, {
+    variables: {
+      resumeText: {
+        step: "trigger",
+        path: "resumeText",
+      },
+    },
+  })
   .then(askAboutSpecialty, {
-    when: { 'gatherCandidateInfo.isTechnical': true },
+    when: { "gatherCandidateInfo.isTechnical": true },
+    variables: {
+      candidateInfo: {
+        step: gatherCandidateInfo,
+        path: ".",
+      },
+    },
   })
   .then(askAboutRole, {
-    when: { 'gatherCandidateInfo.isTechnical': false },
+    when: { "gatherCandidateInfo.isTechnical": false },
+    variables: {
+      candidateInfo: {
+        step: gatherCandidateInfo,
+        path: ".",
+      },
+    },
   });
 
 candidateWorkflow.commit();
 
 (async () => {
   const runResult = await candidateWorkflow.execute({
```

Comment:
the workflow should be passed into `mastra` so it can leverage telemetry and have access to other primitives.